include/torch-mlir/Dialect/Torch/Utils/Utils.h

```diff
@@ -131,6 +131,15 @@ Value createRank0Tensor(PatternRewriter &rewriter, Location loc,
 LogicalResult getTransposedType(BaseTensorType inType, int64_t dimA,
                                 int64_t dimB, Type &transposedType);
 
+// Approximates the heuristic in the torch `acc_type` template for kernels
+// that are defined in terms of it. For now, this just returns accumulators
+// as if for CUDA from that implementation. In the future, this could be
+// extended to look at hints on the `forOp` or its container to better
+// control the behavior. Such support would be done in coordination with
+// the fx_importer and APIs, which could add hints to the IR (based on
+// Torch flags, user options, etc).
+Type getDefaultAccType(PatternRewriter &rewriter, Type inputType);
```

Comment:
Actually this should be specific to a certain op like softmax? We don't want to just convert everything unconditionally? That's what I recall what @stellaraccident recommends--`Type getDefaultAccType(Operation *forOp, Type inputType);`. But i'm certainly not experienced enough here to say exactly.

Reply:
I think it is fine like this : lowerings should use this if the corresponding torch op uses the torch version of this to get a default accumulator type. Still under control of the specific lowering but let's it line up with how torch kernels are structured.

Reply:
(in torch, the soft max kernels defer to this default accumulator heuristic, so we should too)

Reply:
yeah not everything unconditionally, but the same principle applies to convolution and a few other reduction operators (i.e. variance).